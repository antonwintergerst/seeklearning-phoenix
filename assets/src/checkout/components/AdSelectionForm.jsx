import React from 'react';
import { ICONS } from '../../constants';
import Icon from '../../shared/components/Icon';

const AdSelectionForm = ({ adTypes, adFeatures, onSelectClicked }) => (
  <div className="ui basic segment ad-selection-form">
    <table className="ui celled striped table">
      <thead className="center aligned">
        <tr>
          <th />
          {adTypes.map(adType => (
            <th key={adType.id} className={`${adType.isActive ? 'active' : ''}`}>{adType.name}</th>
          ))}
        </tr>
        <tr>
          <th />
          {adTypes.map(adType => (
            <th key={adType.id} className={`${adType.isActive ? 'active' : ''}`}>
              <span className="price">${adType.price}</span>
              <span className="price-suffix">+GST</span>
            </th>
          ))}
        </tr>
        <tr>
          <th />
          {adTypes.map(adType => (
            <th key={adType.id} className={`${adType.isActive ? 'active' : ''}`}>
              <button className={`ui small button${adType.isActive ? ' active' : ''}`} onClick={() => { onSelectClicked(adType); }}>
                Select
              </button>
            </th>
          ))}
        </tr>
      </thead>
      <tbody>
        {adFeatures.map(adFeature => (
          <tr key={adFeature.id}>
            <td>
              <div className="ui ribbon label">{adFeature.title}</div>
            </td>
            {adTypes.map(adType => (
              <td key={adType.id} className={`center aligned${adType.isActive ? ' active' : ''}`}>
                {adType.features.indexOf(adFeature.id) !== -1 &&
                  <Icon icon={ICONS.TICK} />
                }
              </td>
            ))}
          </tr>
        ))}
      </tbody>
      <tfoot className="center aligned">
        <tr>
          <th />
          {adTypes.map(adType => (
            <th key={adType.id}>
              <div className="ui small button">
                Preview
              </div>
            </th>
          ))}
        </tr>
      </tfoot>
    </table>
  </div>
);

export default AdSelectionForm;
