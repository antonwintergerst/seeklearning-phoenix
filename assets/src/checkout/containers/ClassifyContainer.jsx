import React from 'react';
import AdSelectionForm from '../components/AdSelectionForm';
import { adTypes, adFeatures } from './ClassifyContainer.data';

const ClassifyContainer = () => (
  <div>
    <h3 className="ui header">Choose your style of ad</h3>
    <div className="ui stacked segments">
      <AdSelectionForm adTypes={adTypes} adFeatures={adFeatures} />
    </div>
  </div>
);

export default ClassifyContainer;
