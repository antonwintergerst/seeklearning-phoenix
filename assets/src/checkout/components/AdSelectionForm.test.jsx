import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import AdSelectionForm from './AdSelectionForm';
import { adTypes, adFeatures } from '../containers/ClassifyContainer.data';

describe('AdSelectionForm', () => {
  it('renders as expected', () => {
    const tree = ReactTestRenderer.create(<AdSelectionForm adTypes={adTypes} adFeatures={adFeatures} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
