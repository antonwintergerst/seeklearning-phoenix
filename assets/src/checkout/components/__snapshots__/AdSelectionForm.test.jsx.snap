// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdSelectionForm renders as expected 1`] = `
<div
  className="ui basic segment ad-selection-form"
>
  <table
    className="ui celled striped table"
  >
    <thead
      className="center aligned"
    >
      <tr>
        <th />
        <th
          className=""
        >
          Classic
        </th>
        <th
          className=""
        >
          StandOut
        </th>
        <th
          className=""
        >
          Guaranteed Hire
        </th>
      </tr>
      <tr>
        <th />
        <th
          className=""
        >
          <span
            className="price"
          >
            $
            269.99
          </span>
          <span
            className="price-suffix"
          >
            +GST
          </span>
        </th>
        <th
          className=""
        >
          <span
            className="price"
          >
            $
            322.99
          </span>
          <span
            className="price-suffix"
          >
            +GST
          </span>
        </th>
        <th
          className=""
        >
          <span
            className="price"
          >
            $
            394.99
          </span>
          <span
            className="price-suffix"
          >
            +GST
          </span>
        </th>
      </tr>
      <tr>
        <th />
        <th
          className=""
        >
          <button
            className="ui small button"
            onClick={[Function]}
          >
            Select
          </button>
        </th>
        <th
          className=""
        >
          <button
            className="ui small button"
            onClick={[Function]}
          >
            Select
          </button>
        </th>
        <th
          className=""
        >
          <button
            className="ui small button"
            onClick={[Function]}
          >
            Select
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            30 day listing on SEEK
          </div>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            Unlimited changes
          </div>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            Desktop, mobile and tablet
          </div>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            Emailed to jobseekers
          </div>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            3 selling points
          </div>
        </td>
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            Border to stand out
          </div>
        </td>
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            Logo to promote brand
          </div>
        </td>
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            We'll write your ad
          </div>
        </td>
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            Guaranteed placement
          </div>
        </td>
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        >
          <svg
            height="16px"
            style={
              Object {
                "display": "inline-block",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 1024 1024"
            width="16px"
          >
            <path
              d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
              style={
                Object {
                  "fill": "#000",
                }
              }
            />
          </svg>
        </td>
      </tr>
    </tbody>
    <tfoot
      className="center aligned"
    >
      <tr>
        <th />
        <th>
          <div
            className="ui small button"
          >
            Preview
          </div>
        </th>
        <th>
          <div
            className="ui small button"
          >
            Preview
          </div>
        </th>
        <th>
          <div
            className="ui small button"
          >
            Preview
          </div>
        </th>
      </tr>
    </tfoot>
  </table>
</div>
`;
